tools/ctu-analysis/ctu-build.py

```diff
@@ -139,34 +139,30 @@ def generate_ast(params):
             pass
         else:
             raise
-    dir_command = ['cd', directory]
     ast_command = [os.path.join(clang_path, 'clang'), '-emit-ast']
     ast_command.extend(args)
     ast_command.append('-w')
     ast_command.append(source)
     ast_command.append('-o')
     ast_command.append(ast_path)
-    ast_command_str = ' '.join(dir_command) + " && " + ' '.join(ast_command)
-    logging.info(ast_command_str)
-    subprocess.call(ast_command_str, shell=True)
+    os.chdir(directory)
+    logging.info(ast_command)
+    subprocess.call(ast_command, shell=False)
 
 
 def map_functions(params):
     command, sources, directory, clang_path, ctuindir = params
     args = get_command_arguments(command)
-    logging.info("map_functions command: " + command)
-    logging.info("sources: " + ' '.join(sources))
     arch = get_triple_arch(clang_path, args, sources[0])
-    dir_command = ['cd', directory]
     funcmap_command = [os.path.join(clang_path, 'clang-func-mapping')]
     funcmap_command.extend(sources)
     funcmap_command.append('--')
-    funcmap_command.extend(args)
-    funcmap_command_str = ' '.join(dir_command) + \
-                          " && " + ' '.join(funcmap_command)
-    logging.info("Calling function map: " + funcmap_command_str)
+    funcmap_command.extend(args)    
     output = []
-    fn_out = subprocess.check_output(funcmap_command_str, shell=True)
+    os.chdir(directory);
+    logging.info("map_functions: ")
+    logging.info(funcmap_command)
+    fn_out = subprocess.check_output(funcmap_command)
```

Comment:
Have you tried some of our old projects with escape problem prone constructions? I think changing shell=True from shell=False can have an influence on escaping the whole command line, so we can screw up some stuff. I think openssl was heavily affected by the escaping problem.